CRM/MembershipExtras/Job/UpdateMembershipPeriod.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class CRM_MembershipExtras_Job_UpdateMembershipPeriod {
+
+  /**
+   * @var array
+   */
+  private $overdueContributionStatusIDs = [];
+  
+  /**
+   * Class constructor.
+   */
+  public function __construct() {
+    $this->setOverdueContributionStatusIDs();
+  }
+
+  private function setOverdueContributionStatusIDs() {
+    $contributionStatusesNameMap = CRM_MembershipExtras_Service_ContributionUtilities::getContributionStatusesNameMap();
+    $pendingStatusID = $contributionStatusesNameMap['Pending'];
```

Comment:
have you checked with Guanhuan that these are the statuses that should be considered overdue and nothing else ?

Reply:
Camilo advised on it.